File: solutions/clojure/diamond/1/src/diamond.clj
(ns diamond)

(defn make_spaces [k]
  (apply str (repeat k " "))
  )

(defn make_line [[k l chr]]
  (if (zero? l)
    (str (make_spaces k) chr (make_spaces k))
    (str
     (make_spaces k) chr (make_spaces (dec (* l 2))) chr (make_spaces k))
    )
  )

(defn make_level_line [[width level]] 
  (make_line [(- width level) level (char (+ (int \A ) level))] )
  )

(defn diamond [letter] 
  (let [d (- (int letter) (int \A))]
    (for [idx (range (- d) (inc d))]
        (make_level_line [d (- d (abs idx))] )
      )
    )    
)

File: solutions/python/diamond/1/diamond.py
def rows(letter):
    rows = []
    m = ord(letter[0])- ord('A')
    nr = 2*(m+1)-1
    for i in range(nr):
        current_row = ""
        idx = i
        if i>m:
            idx = - idx + 2*m
        current_char = chr(idx + ord('A'))
        for j in range(-m,m+1):
            if j == idx or j == -idx:
                current_row += current_char
            else:
                current_row += ' '
        rows.append(current_row)
    return rows
